queue/abstract.lua

```diff
@@ -368,6 +407,12 @@ local function make_self(driver, space, tube_name, tube_type, tube_id, opts)
     }, {
         __index = tube
     })
+
+    if checked_drivers[tube_type] == nil then
+        check_driver_api(self.raw, tube_type)
+        checked_drivers[tube_type] = true
+    end
```

Comment:
How about doing this check inside `register_driver()` function? It already contains check for presence of a couple of driver functions.

I see, built-in drivers are registered using just `require()`, but nothing stops us from changing it to explicit calls of `queue.register_driver()`.

Okay, I see, there is another problem: two implementations of `register_driver()`, but I think we can just keep one in `init.lua` and zap one in `abstract.lua` with a comment that it is defined in `init.lua`.

BTW, it takes the question above about hot reload off the table.

Reply:
Hm, I see that you already zapped `register_driver()` from `abstract.lua` within PR #130, but you still don't use the method to verify driver functions and postpone the check until a tube (re)creation. I don't see a reason for this.

Reply:
> Hm, I see that you already zapped `register_driver()` from `abstract.lua` within PR #130, but you still don't use the method to verify driver functions and postpone the check until a tube (re)creation. I don't see a reason for this.

How? I can't check the driver methods before creating it. In `register_driver()` I can only check the `tube` methods.

Reply:
I see it now: `register_driver()` checks driver module functions, while here you add checks for driver instance methods.